packages/serverless-cdk-plugin/src/utils/resolveVariablesInCdkOutput.ts

```diff
@@ -0,0 +1,47 @@
+import traverse from 'traverse';
+
+import { CloudFormationTemplate, ResolveVariable } from 'types';
+
+const resolveVariablesInString = async (
+  resolveVariable: ResolveVariable,
+  stringToResolve: string,
+) => {
+  try {
+    const resolvedString = await resolveVariable(stringToResolve);
+
+    // If variable was resolved in the string, the return value is ${myResolvedValue}
+    if (!resolvedString.match(/\$\{.*\}/)) {
+      throw new Error('Unexpected returned value from sls variable resolver');
+    }
+
+    return resolvedString.slice('${'.length, -'}'.length);
+  } catch {
+    // The resolving failed, we silently return the original string
+    return stringToResolve;
+  }
+};
+
+export const resolveVariablesInCdkOutput = async (
+  resolveVariable: ResolveVariable,
+  cdkOutput: CloudFormationTemplate,
+): Promise<void> => {
+  const toResolvePaths: string[][] = [];
+
+  const traversedOutput = traverse(cdkOutput);
+  traversedOutput.forEach(function (node: unknown) {
+    if (typeof node === 'string' && node.includes('${')) {
+      toResolvePaths.push(this.path);
```

Comment:
ca peut être intéressant de dire ce que ca fait this.path dans une arrow function dans un comment.

ca demande pas mal de temps de comprendre que le `this` transcende le contexte et que ca denote le path du file